website/views.py

```diff
@@ -0,0 +1,142 @@
+#! /usr/bin/env python
+
+
+from flask import Blueprint, render_template, session, request, redirect, url_for
+from . import interfacer, loginManager, userManager, tagManager, gameManager
+from . import constants as const
+views = Blueprint('views',__name__)
+
+
+@views.route('/')
+def home():
+    return render_template("home.html")
+
+
+@views.route('/users', methods=['GET', 'POST'])
+def user():
+    if request.method == "POST":
+        viewerID = int(request.form.get('viewerID',''))
+        if(session['userID'] == viewerID):
```

Comment:
Since this is the bit of code that you _always_ use to check for permission, you can extract this out into a wrapper! Python wrappers are the @ symbols before the functions, like `@views.route` & are Higher Order Functions: they take a function and return a modified version.
Try something like:

```py
def login_required(f):
    def inner(*args, **kwargs):
        if (session['userID'] == viewerID):
            return f(*args, **kwargs)
        else:
            flash("Fucko!")
            return redirect('views.login', code=302)
    return inner
```